extensions/amp-story/1.0/amp-story-request-service.js

```diff
@@ -56,12 +56,33 @@ export class AmpStoryRequestService {
   /**
    * Retrieves the publisher share providers.
    * Has to be called through `loadShareConfig`.
+   * @param  {!Element} shareConfigEl
    * @return {(!Promise<!JsonObject>|!Promise<null>)}
    */
-  loadShareConfigImpl_() {
-    const shareConfigEl = this.storyElement_.querySelector(
-      'amp-story-social-share, amp-story-bookend'
-    );
+  getInlineConfig(shareConfigEl) {
+    // Fallback. Check for an inline json config.
+    let config = null;
+    try {
+      config = getChildJsonConfig(shareConfigEl);
+    } catch (err) {}
+
+    return Promise.resolve(config);
+  }
+
+  /**
+   * Retrieves the publisher share providers.
+   * Has to be called through `loadShareConfig`.
+   * @param  {!Element} element
+   * @return {(!Promise<!JsonObject>|!Promise<null>)}
+   * @public
+   */
+  loadConfigImpl(element) {
+    const shareConfigEl = element
+      ? element
+      : this.storyElement_.querySelector(
+          'amp-story-social-share, amp-story-bookend'
+        );
+
     if (!shareConfigEl) {
       return Promise.resolve();
     }
```

Comment:
The `element` parameter currently has the non-null (!) type, meaning that it is always expected to be defined here. If the passed in `element` is truly never null, then this code isn't necessary because `shareConfigEl` will always be successfully set to `element`. But if `loadConfigImpl()` is sometimes called with a potentially null `element` value, then you instead need to update the `element` parameter's type to `{?Element}`